backend/file-ucloud-service/src/jvmMain/kotlin/services/SyncthingClient.kt

```diff
@@ -443,28 +443,316 @@ class SyncthingClient(
+                                    """
+                            )
+                        }.rows
+
+                        pendingDevices.forEach { device ->
+                            val newFolders = result
+                                .filter {
+                                    it.getString("local_device_id") == device.id
+                                }
+                                .distinctBy { it.getField(SyncFoldersTable.id) }
+                                .map { row ->
+                                    SyncthingFolder(
+                                        id = row.getField(SyncFoldersTable.id).toString(),
+                                        label = row.getField(SyncFoldersTable.path).substringAfterLast("/"),
+                                        devices = result
+                                            .filter {
+                                                it.getString("local_device_id") == device.id &&
+                                                    it.getField(SyncFoldersTable.id) == row.getField(
+                                                    SyncFoldersTable.id
+                                                ) &&
+                                                    it.getField(SyncDevicesTable.user) == row.getField(
+                                                    SyncFoldersTable.user
+                                                )
+                                            }.map {
+                                                SyncthingFolderDevice(it.getField(SyncDevicesTable.device))
+                                            },
+                                        path = File(
+                                            "/mnt/sync",
+                                            row.getField(SyncFoldersTable.id).toString()
+                                        ).absolutePath,
+                                        type = SynchronizationType.valueOf(row.getField(SyncFoldersTable.syncType)).syncthingValue,
+                                        rescanIntervalS = device.rescanIntervalSeconds
+                                    )
+                                }
+
+                            try {
+                                val resp = httpClient.put<HttpResponse>(deviceEndpoint(device, "/rest/config/folders")) {
+                                    body = TextContent(
+                                        defaultMapper.encodeToString(newFolders),
+                                        ContentType.Application.Json
+                                    )
                                     headers {
                                         append("X-API-Key", device.apiKey)
                                     }
                                 }
 
-                                if (resp.status == HttpStatusCode.OK) {
+                                if (resp.status != HttpStatusCode.OK) {
```

Comment:
Technically speaking this could reject requests which were successful but didn't return explicitly 200 (e.g. anything between 201-299). Seems unlikely though.